### ISSUE TITLE
History is not working as expected

### ISSUE THREAD

**Issue by VikramDudheria91** [jarrodek/ChromeRestClient#703]
Vikram Dudheria

Saved Requests in history should not over write themselves with the current one.
Meaning Both requests should be there i.e 
old request and the latest request

Saved Requests in history are getting over written with the current one.

## Versions
App: 7.20.37.203-stable
Chrome: 50.0.2661.102
Platform: Windows (7)

## Steps to reproduce
1. 
2. 
3.

**Comment by jarrodek** [comment]
Hi,

Thank you for issue report.

It is working as intended though. When saved request is restored from saved list it will recall the state from the time when the request was saved. However, if you recall the same request from history list it will restore the last used settings.
I'd be inclined to close this as WAI.

**Comment by VikramDudheria91** [comment]
Hi,
But in the earlier version it was their that if you use the saved request from history it will create the requests at the current timestamp and also the earlier will also be there.

Sent from Yahoo Mail on Android 
 
  On Tue, Jun 14, 2016 at 3:05 PM, Paweł Psztyć<notifications@github.com> wrote:   
Hi,

Thank you for issue report.

It is working as intended though. When saved request is restored from saved list it will recall the state from the time when the request was saved. However, if you recall the same request from history list it will restore the last used settings.
I'd be inclined to close this as WAI.

**Comment by jarrodek** [comment]
Yes, and this is the same as current version is doing.
The history request is recognizable by HTTP method and the URL. If the object matching it is in history it will append current state to the history.
I guess you are talking about the case when you change other parameters (like headers or body). It's not the case anymore and it intentionally isn't. It made the history less readable and messy and searching through the history was a painful experience. 
The app keeps the history of parameters for each call so it is possible to recover previous state. However there's no UI for that. Some time ago I started coding it but the change was to big to finish it at the time so I decided to work on it in the future. It's not a priority though.
